Sources/SecureXPC/Server/XPCClientRequirement.swift

```diff
@@ -8,88 +8,364 @@
+            guard let teamID = try teamIdentifierForThisProcess() else {
+                throw XPCError.misconfiguredServer(description: "This server does not have a team identifier")
+            }
+            
+            return try teamIdentifier(teamID)
+        }
+    }
     
-    /// The requesting client must be within the same parent app bundle as this server.
+    /// The requesting client must be within the same parent bundle as this server.
     ///
-    /// If this server is not part of an app bundle, then this always evaluates to false.
-    case sameAppBundle
+    /// - Throws: If this server is not part of a bundle.
+    public static var sameParentBundle: XPCClientRequirement {
+        get throws {
+            XPCClientRequirement(messageAcceptor: ParentBundleMessageAcceptor(parentBundleURL: try parentBundleURL))
+        }
+    }
     
-    /// The requesting client must satisfy the designated requirement of the parent app bundle.
+    /// The requesting client must satisfy the designated requirement of the parent bundle.
     ///
-    /// If this server has no parent app bundle, then this always evaluates to false.
-    case parentAppDesignatedRequirement
-    
-    /// The requesting client must be running in the same process as this server.
-    case sameProcess
+    /// - Throws: If this server has no parent bundle.
+    public static var parentDesignatedRequirement: XPCClientRequirement {
+        get throws {
+            let parentBundleURL = try parentBundleURL
+            var parentCode: SecStaticCode?
+            var parentRequirement: SecRequirement?
+            guard SecStaticCodeCreateWithPath(parentBundleURL as CFURL, [], &parentCode) == errSecSuccess,
+                  let parentCode = parentCode,
+                  SecCodeCopyDesignatedRequirement(parentCode, [], &parentRequirement) == errSecSuccess,
+                  let parentRequirement = parentRequirement else {
+                throw XPCError.internalFailure(description: "Unable to determine designated requirement for parent " +
+                                                            "bundle: \(parentBundleURL)")
+            }
+            
+            return secRequirement(parentRequirement)
+        }
+    }
     
     /// The requesting client must satisfy both requirements.
-    case and(XPCClientRequirement, XPCClientRequirement)
+    public static func and(_ lhs: XPCClientRequirement, _ rhs: XPCClientRequirement) -> XPCClientRequirement {
+        XPCClientRequirement(messageAcceptor: AndMessageAcceptor(lhs: lhs.messageAcceptor, rhs: rhs.messageAcceptor))
+    }
     
     /// The requesting client must satisfy at least one of the requirements.
-    case or(XPCClientRequirement, XPCClientRequirement)
+    public static func or(_ lhs: XPCClientRequirement, _ rhs: XPCClientRequirement) -> XPCClientRequirement {
+        XPCClientRequirement(messageAcceptor: OrMessageAcceptor(lhs: lhs.messageAcceptor, rhs: rhs.messageAcceptor))
+    }
```

Comment:
I think it's be pretty nice/natural/expressive to use `&&` and `||` operators for this. It would let people have bracket-less, infix notatio

Like: `a && b && c && d`
Instead of: `and(a, and(b, and(c, d)`


Alternatively, `and` `or` could be replaced with `all()` and `any`. This might have some performance benefit w.r.t. allocating fewer existential containers as compared to a tree of `AndMessageAcceptor`/`OrMessageAcceptor` instances.

Reply:
Ah yeah that could be nice, hadn't thought of that. Let me try it out.

Reply:
Yeah the custom operator implementations are nice, will change to using them.